omod-1.11/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_11/OrderableResource1_11.java

```diff
@@ -0,0 +1,198 @@
+import org.openmrs.Concept;
+import org.openmrs.ConceptClass;
+import org.openmrs.ConceptName;
+import org.openmrs.ConceptSearchResult;
+import org.openmrs.OrderType;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.PropertyGetter;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.RefRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.Searchable;
+import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+/**
+ * {@link org.openmrs.module.webservices.rest.web.annotation.Resource} for
+ * {@link org.openmrs.ConceptSearchResult}, this resource is used to search for concepts that are
+ * orderable. It returns ConceptSearchResults instead of Concepts
+ */
+@Resource(name = RestConstants.VERSION_1 + "/orderable", supportedClass = Concept.class, supportedOpenmrsVersions = {
+        "1.11.*", "1.12.*", "2.0.*" })
+public class OrderableResource1_11 extends BaseDelegatingResource<ConceptSearchResult> implements Searchable {
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#newDelegate()
+	 */
+	@Override
+	public ConceptSearchResult newDelegate() {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#save(Object)
+	 */
+	@Override
+	public ConceptSearchResult save(ConceptSearchResult delegate) {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+	
+	@Override
+	public SimpleObject search(RequestContext context) throws ResponseException {
+		
+		String query = context.getParameter("q");
+		
+		List<OrderType> orderTypes = Context.getOrderService().getOrderTypes(false);
+		List<ConceptClass> orderableConceptClasses = getOrderableConceptClasses(orderTypes);
+		
+		List<ConceptClass> conceptClasses = null;
+		String[] classUuids = context.getRequest().getParameterValues("conceptClasses");
+		
+		if (classUuids != null) {
+			for (String uuid : classUuids) {
+				ConceptClass cc = getConceptClassByUuid(uuid, orderableConceptClasses);
```

Comment:
Switched to this. I was using this function to search for a concept class from the already loaded concept classes instead of loading the class from the database again.